src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -5,14 +5,30 @@ import mesosphere.marathon.state.AppDefinition
 import mesosphere.marathon.tasks.PortsMatcher
 import mesosphere.mesos.protos.{ RangesResource, Resource }
 import org.apache.log4j.Logger
+import org.apache.mesos.Protos
 import org.apache.mesos.Protos.Offer
 
 import scala.collection.JavaConverters._
 import scala.collection.immutable.Seq
+import scala.collection.mutable
 
 object ResourceMatcher {
   type Role = String
 
+  case class ScalarMatch(requiredValue: Double, offeredValue: Double, role: String) {
```

Comment:
How about making `role` an `Option[String]` instead?  Empty string won't be a valid, so this pushes some complexity to the users of this structure.